lib/r10k/forge/module_release.rb

```diff
@@ -108,12 +115,70 @@ def download
       def verify
         logger.debug1 "Verifying that #{@tarball_cache_path} matches checksum"
 
-        md5_of_tarball = Digest::MD5.hexdigest(File.read(@tarball_cache_path, mode: 'rb'))
+        sha256_of_tarball = Digest::SHA256.file(@tarball_cache_path).hexdigest
+
+        if @sha256_file_path.exist?
+          verify_from_sha256_file(sha256_of_tarball)
+        else
+          begin
+            verify_sha256_from_forge(sha256_of_tarball)
+          rescue R10K::Error
+            if R10K::Util::Platform.fips?
+              raise R10K::Error, "Could not verify module, no SHA256 checksum available, and MD5 checksums not allowed in FIPS mode"
+            end
+
+            logger.debug1 "No SHA256 checksum available, falling back to MD5"
+
+            md5_of_tarball = Digest::MD5.file(@tarball_cache_path).hexdigest
+
+            if @md5_file_path.exist?
+              verify_from_md5_file(md5_of_tarball)
```

Comment:
If we parameterized these methods could we condense them? It seems that they all do largely the same thing but are reading different instance variables (and logging a formulaic, but slightly different message).

Reply:
Yeah I was thinking about that. I'll look into it.